### ISSUE TITLE
When will the checkpoint be made public?

### ISSUE THREAD

**Issue by qqtang-code** [princeton-pli/PruLong#5]
Great Work!
I found some problems when reproducing PruLong. When I downloaded prolongation-512K from huggingface and processed it with scripts/unpack_dataset.py, I found that the model training was not convergent.

<img width="1280" height="1137" alt="Image" src="https://github.com/user-attachments/assets/0cb566ec-2faa-47e9-8207-340a7d846929" />

But when I processed it with concat_packed_datasets.py, it felt like it was back to normal.

<img width="5970" height="5307" alt="Image" src="https://github.com/user-attachments/assets/082a9a70-4bfa-429e-8a5b-cc4b43062da7" />

Could you explain the reason?

I would be very grateful if you could reply to me in time. If you need to provide other details, please also let me know in this issue.

**Comment by testzer0** [comment]
Hi, thanks for bringing this to our attention! Can you point out the specific checkpoint link and the hyperparameters (or the entire script, if possible) that you are using? The model training sparsity plot looks right here, but the regularization loss suggests that perhaps the LR for the alphas was lower than necessary (when this is true -> at first the sparsity does not increase at all, then at some point the loss is too large and it overcorrects). I can take a closer look with some more details.

Adithya

**Comment by qqtang-code** [comment]
> Hi, thanks for bringing this to our attention! Can you point out the specific checkpoint link and the hyperparameters (or the entire script, if possible) that you are using? The model training sparsity plot looks right here, but the regularization loss suggests that perhaps the LR for the alphas was lower than necessary (when this is true -> at first the sparsity does not increase at all, then at some point the loss is too large and it overcorrects). I can take a closer look with some more details.嗨，感谢你引起我们的注意！你能指出你使用的具体检查点链接和超参数（如果可能，请提供整个脚本）吗？模型训练的稀疏性图看起来是正确的，但正则化损失表明，alpha 的 LR 可能低于必要水平（当这种情况发生时 -> 最初稀疏性根本不会增加，然后在某个点上损失太大并过度校正）。我可以提供更多细节后更仔细地查看。
> 
> Adithya

The Model adopted was Llama3.1-8B-instruct, and the script used was the one provided in your repo. No modifications were made, only changes were made to the data part. The upper figure uses proprolong -512k-mix, but performs the scripts/unpack_dataset.py operation on the data (because I didn't know what data format to organize at the beginning), and the lower figure performs the concat_packed_datasets.py operation.
This is the complete representation of the following picture：

<img width="1031" height="898" alt="Image" src="https://github.com/user-attachments/assets/4527fdef-34a1-423e-97dd-ec7c30bfb2ee" />

**Comment by testzer0** [comment]
Interesting! I will look into this issue more, but off the top of my head, I think this might arise because by default we train using packed sequences that have input ids of the shape `[total_sequence_length]`, or equivalently, don't have a batch dimension. When you use unpacked sequences, you are effectively using a lower batch size since each sequence only has one sequence packed. I suspect this might make the optimal hyperparameters different. In any case, we are currently running some additional experiments and plan to release a few checkpoints in a few weeks once those are done. Hopefully, those should provide a more direct way of running the method!

**Comment by qqtang-code** [comment]
> Interesting! I will look into this issue more, but off the top of my head, I think this might arise because by default we train using packed sequences that have input ids of the shape `[total_sequence_length]`, or equivalently, don't have a batch dimension. When you use unpacked sequences, you are effectively using a lower batch size since each sequence only has one sequence packed. I suspect this might make the optimal hyperparameters different. In any case, we are currently running some additional experiments and plan to release a few checkpoints in a few weeks once those are done. Hopefully, those should provide a more direct way of running the method!

I'm really looking forward to your release.
